lib/features/ai/data/datasource_impl/ai_datasource_impl.dart

```diff
@@ -0,0 +1,56 @@
+import 'package:dio/dio.dart';
+import 'package:tmail_ui_user/features/ai/data/datasource/ai_datasource.dart';
+import 'package:tmail_ui_user/features/ai/domain/model/ai_response.dart';
+
+class AIDataSourceImpl implements AIDataSource {
+  final Dio _dio;
+  final String _apiKey = 'REDACTED';
+  final String _baseUrl = 'https://ai.linagora.com/api/v1';
+  final String _model = 'gpt-oss-120b';
+
+  AIDataSourceImpl({
+    required Dio dio,
+  }) : _dio = dio;
+  @override
+  Future<AIResponse> request(String prompt) async {
+    try {
+      final response = await _dio.post(
```

Comment:
Create another class to handle interactions with the api and the datasource calls it.